nall/GNUmakefile

```diff
@@ -187,12 +198,20 @@ endif
 
 # windows settings
 ifeq ($(platform),windows)
-  options += -mthreads -lpthread -lws2_32 -lole32
+  extension := .exe
+  ifeq ($(msvc),true)
+    flags += -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_WARNINGS -DNOMINMAX
+  endif
+  options += -mthreads -lws2_32 -lole32 -lshell32
   options += $(if $(findstring clang++,$(compiler)),-fuse-ld=lld)
-  options += $(if $(findstring g++,$(compiler)),-static -static-libgcc -static-libstdc++)
```

Comment:
A couple of things to note:
- Due to "g++" being a substring of "clang++", this was being applied to both compilers.
- On my machine, -static-libgcc and -static-libstdc++ weren't having any effect on the build output with either GCC or Clang. I removed them because they caused unused option warnings in the new build configuration.